// Copyright 2019 Zeppelin Bend Pty Ltd
// This file is part of cimbend.
// 
// cimbend is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// cimbend is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with cimbend.  If not, see <https://www.gnu.org/licenses/>.


syntax = "proto3";
option java_multiple_files = true;
option java_package = "com.zepben.protobuf.cim.iec61970";
package zepben.cim.iec61970;

message DiagramObject {
    string name = 1;
    string mRID = 2;
    string IdentifiedObjectMRID = 3; // Only used if sending DiagramObjectPoint's separately to equipment.
    repeated DiagramObjectPoint diagramObjectPoints = 4; // Only used if sending DiagramObjectPoint's separately to equipment.
}

/* A point in a given space defined by 3 coordinates and associated to a diagram object.
The coordinates may be positive or negative as the origin does not have to be in the corner of a diagram.*/
message DiagramObjectPoint {
    double xPosition = 1;
    double yPosition = 2;
}
